Server crash when using kc.sh with -Dkeycloak.profile=experimental

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

While experimenting with the -Dkeycloak.profile=experimental feature in Keycloak 25, the server crashes upon startup.

### Version

25.0.0

### Regression

- [ ] The issue is a regression

### Expected behavior

The Keycloak server should start successfully with the experimental profile enabled, or provide a warning that the parameter is not supported.

### Actual behavior

The server crashes and fails to start when provided with the -Dkeycloak.profile=experimental argument.

### How to Reproduce?

1. Download Keycloak 25.
2. Execute the following command.
```
bin/kc.sh -Dkeycloak.profile=experimental start-dev --http-port 8180 --https-port=8443
```

### Anything else?

**Log error:**

```
2024-06-14 13:39:02,127 DEBUG [io.smallrye.config._private] (main) SRCFG01006: Loaded ConfigSource DefaultValuesConfigSource with ordinal -2147483648

Exception in thread "main" java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Caused by: java.lang.IllegalArgumentException: No enum constant org.keycloak.common.Profile.ProfileName.EXPERIMENTAL
        at java.base/java.lang.Enum.valueOf(Enum.java:273)
        at org.keycloak.common.Profile$ProfileName.valueOf(Profile.java:412)
        at org.keycloak.common.profile.PropertiesProfileConfigResolver.getProfileName(PropertiesProfileConfigResolver.java:24)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:1002)
        at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
        at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
        at org.keycloak.common.Profile.configure(Profile.java:228)
        at org.keycloak.quarkus.runtime.Environment.getCurrentOrCreateFeatureProfile(Environment.java:256)
        at org.keycloak.quarkus.runtime.configuration.mappers.PropertyMappers$MappersConfig.lambda$sanitizeDisabledMappers$1(PropertyMappers.java:261)
        at org.keycloak.quarkus.runtime.configuration.DisabledMappersInterceptor.runWithDisabled(DisabledMappersInterceptor.java:86)
        at org.keycloak.quarkus.runtime.configuration.mappers.PropertyMappers$MappersConfig.sanitizeDisabledMappers(PropertyMappers.java:255)
        at org.keycloak.quarkus.runtime.configuration.mappers.PropertyMappers.sanitizeDisabledMappers(PropertyMappers.java:125)
        at org.keycloak.quarkus.runtime.cli.Picocli.parseAndRun(Picocli.java:118)
        at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:99)
        ... 6 more
```

--- a/common/src/main/java/org/keycloak/common/profile/PropertiesProfileConfigResolver.java
+++ b/common/src/main/java/org/keycloak/common/profile/PropertiesProfileConfigResolver.java
@@ -21,7 +21,15 @@ public class PropertiesProfileConfigResolver implements ProfileConfigResolver {
     @Override
     public Profile.ProfileName getProfileName() {
         String profile = getter.apply("keycloak.profile");
-        return profile != null ? Profile.ProfileName.valueOf(profile.toUpperCase()) : null;
+
+        if (profile != null) {
+            try {
+                return Profile.ProfileName.valueOf(profile.toUpperCase());
+            } catch (IllegalArgumentException e) {
+                throw new ProfileException(String.format("Invalid profile '%s' specified via 'keycloak.profile' property", profile));
+            }
+        }
+        return null;
     }
 
     @Override

--- a/common/src/test/java/org/keycloak/common/ProfileTest.java
+++ b/common/src/test/java/org/keycloak/common/ProfileTest.java
@@ -85,11 +85,9 @@ public class ProfileTest {
         Properties properties = new Properties();
         properties.setProperty("keycloak.profile.feature.account_api", "disabled");
 
-        try {
-            Profile.configure(new PropertiesProfileConfigResolver(properties));
-        } catch (ProfileException e) {
-            Assert.assertEquals("Feature account3 depends on disabled feature account-api", e.getMessage());
-        }
+        Assert.assertEquals("Feature account3 depends on disabled feature account-api",
+                assertThrows(ProfileException.class,
+                        () -> Profile.configure(new PropertiesProfileConfigResolver(properties))).getMessage());
     }
 
     @Test
@@ -107,11 +105,19 @@ public class ProfileTest {
         Properties properties = new Properties();
         properties.setProperty("keycloak.profile.feature.account_api", "invalid");
 
-        try {
-            Profile.configure(new PropertiesProfileConfigResolver(properties));
-        } catch (ProfileException e) {
-            Assert.assertEquals("Invalid config value 'invalid' for feature key keycloak.profile.feature.account_api", e.getMessage());
-        }
+        Assert.assertEquals("Invalid config value 'invalid' for feature key keycloak.profile.feature.account_api",
+                assertThrows(ProfileException.class,
+                        () -> Profile.configure(new PropertiesProfileConfigResolver(properties))).getMessage());
+    }
+
+    @Test
+    public void wrongProfileInProperties() {
+        Properties properties = new Properties();
+        properties.setProperty("keycloak.profile", "experimental");
+
+        Assert.assertEquals("Invalid profile 'experimental' specified via 'keycloak.profile' property",
+                assertThrows(ProfileException.class,
+                        () -> Profile.configure(new PropertiesProfileConfigResolver(properties))).getMessage());
     }
 
     @Test

